tests/component/test_task_events.py

```diff
@@ -0,0 +1,414 @@
+    ao_task.register_every_n_samples_transferred_from_buffer_event(
+        100, event_observer.handle_every_n_samples_event
+    )
+    ao_task.timing.cfg_samp_clk_timing(
+        rate=10000.0, sample_mode=AcquisitionType.FINITE, samps_per_chan=1000
+    )
+
+    with pytest.raises(nidaqmx.DaqError) as exc_info:
+        ao_task.register_every_n_samples_transferred_from_buffer_event(
+            100, event_observer.handle_every_n_samples_event
+        )
+
+    assert (
+        exc_info.value.error_code
+        == DAQmxErrors.EVERY_N_SAMPS_TRANSFERRED_FROM_BUFFER_EVENT_ALREADY_REGISTERED
+    )
+
+
+@pytest.mark.xfail(
+    reason="AB#2395958: GrpcStubInterpreter does not report event registration errors to caller"
+)
+def test___signal_event_registered___register_signal_event___already_registered_error_raised(
+    ai_task: nidaqmx.Task,
+) -> None:
+    event_observer = SignalEventObserver()
+    ai_task.register_signal_event(Signal.SAMPLE_COMPLETE, event_observer.handle_signal_event)
+    ai_task.timing.cfg_samp_clk_timing(
+        rate=10000.0, sample_mode=AcquisitionType.FINITE, samps_per_chan=1000
+    )
+
+    with pytest.raises(nidaqmx.DaqError) as exc_info:
+        ai_task.register_signal_event(Signal.SAMPLE_COMPLETE, event_observer.handle_signal_event)
+
+    assert exc_info.value.error_code == DAQmxErrors.SIGNAL_EVENT_ALREADY_REGISTERED
+
+
+@pytest.mark.xfail(
+    reason="AB#2395958: GrpcStubInterpreter does not report event registration errors to caller"
+)
+def test___ai_task___register_wrong_every_n_samples_event___not_supported_by_device_error_raised(
+    ai_task: nidaqmx.Task,
+) -> None:
+    event_observer = EveryNSamplesEventObserver()
+    ai_task.timing.cfg_samp_clk_timing(
+        rate=10000.0, sample_mode=AcquisitionType.FINITE, samps_per_chan=1000
+    )
+
+    with pytest.raises(nidaqmx.DaqError) as exc_info:
+        ai_task.register_every_n_samples_transferred_from_buffer_event(
+            100, event_observer.handle_every_n_samples_event
+        )
+
+    assert (
+        exc_info.value.error_code
+        == DAQmxErrors.EVERY_N_SAMPS_TRANSFERRED_FROM_BUFFER_EVENT_NOT_SUPPORTED_BY_DEVICE
+    )
+
+
+@pytest.mark.xfail(
+    reason="AB#2395958: GrpcStubInterpreter does not report event registration errors to caller"
+)
+def test___ao_task___register_wrong_every_n_samples_event___not_supported_by_device_error_raised(
+    ao_task: nidaqmx.Task,
+) -> None:
+    event_observer = EveryNSamplesEventObserver()
+    ao_task.timing.cfg_samp_clk_timing(
+        rate=10000.0, sample_mode=AcquisitionType.FINITE, samps_per_chan=1000
+    )
+
+    with pytest.raises(nidaqmx.DaqError) as exc_info:
+        ao_task.register_every_n_samples_acquired_into_buffer_event(
+            100, event_observer.handle_every_n_samples_event
+        )
+
+    assert (
+        exc_info.value.error_code
+        == DAQmxErrors.EVERY_N_SAMPLES_ACQ_INTO_BUFFER_EVENT_NOT_SUPPORTED_BY_DEVICE
+    )
```

Comment:
I added these tests because we could reasonably catch double registration in Python and return the correct error, but we can't catch all of the other registration errors that DAQmx might possibly return.